syntax = "proto2";
message InitWorld{
required int64 worldid = 1;
required int64 seqnum=2;
}
message WarehouseInfo{
required int32 whid = 1;
required int32 wh_x = 2;
required int32 wh_y = 3;
}
message DetermineWarehouse{
repeated WarehouseInfo warehouses=1;
required int64 seqnum = 2;
}
message DetermineTruck{
repeated TruckLocation arrivedtrucks = 1;
required int64 seqnum = 2;
}
message TruckLocation{
required int32 truckid = 1;
required int32 whid = 2;
}
message DetermineDst{
repeated TruckDst leavingtrucks = 1;
required int64 seqnum = 2;
}
message TruckDst{
required int32 truckid = 1;
required int32 x = 2;
required int32 y = 3;
required int64 shipid = 4;
}
message SettleShipment{
repeated PackageInfo packages = 1;
required int64 seqnum = 2;
}
message PackageInfo{
required int64 packageid = 1;
required int64 shipid = 2;
}
message UCommands{
repeated InitWorld worlds = 1;
repeated DetermineTruck trucks = 2;
repeated SettleShipment shipments = 3;
}
message ACommands{
repeated DetermineWarehouse warehouses = 1;
repeated DetermineDst dests = 2;
}
message Response{
required bool succeed = 1;
required int64 seqnum = 2;
}